### PR TITLE
feat: Redesigned corporate card swiper in dashboard to latest designs

### PR DESCRIPTION
**Pull request by OmkarJ13**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=-5D68Li)
### <samp>🤖 Generated by Copilot at 51f377e</samp>

This pull request refactors the corporate card feature to use a new model, service, and components to display the card details and stats. It also fixes some UI issues and improves the performance and readability of the code. It moves the `CorporateCardComponent` to the shared module and renames some files accordingly. It updates the `CardStatsComponent`, `CardDetailComponent`, and `SpentCardsComponent` to use the new `NewCardDetail` model and `CorporateCreditCardExpenseService`. It also modifies some templates and styles to adjust the height and visibility of the card elements.

### <samp>🤖 Generated by Copilot at 51f377e</samp>

> _To show corporate cards and their stats_
> _We refactored some components and classes_
> _We used `NewCardDetail`_
> _And removed what was stale_
> _And made `CorporateCardComponent` more sassy_

### <samp>🤖 Generated by Copilot at 51f377e</samp>

*  Add a new interface `NewCardDetail` to represent the details and stats of a corporate card ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-123f6efc7202ed91275b13f8458dd26b892c9ba8f822660f63d274f9c5708782R1-R13))
*  Remove the unused import of `CardDetail` from `card-stats.component.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-fcc80bf83d226889337842b0e4ab887285aa2d2383c75f304b5d1b6640345fa9L5))
*  Import `CorporateCreditCardExpenseService` and `NewCardDetail` to `card-stats.component.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-fcc80bf83d226889337842b0e4ab887285aa2d2383c75f304b5d1b6640345fa9R11-R12))
*  Change the type of `cardTransactionsAndDetails$` from `Observable<CardDetail[]>` to `Observable<NewCardDetail[]>` in `card-stats.component.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-fcc80bf83d226889337842b0e4ab887285aa2d2383c75f304b5d1b6640345fa9L19-R20))
*  Inject `CorporateCreditCardExpenseService` to the constructor of `CardStatsComponent` in `card-stats.component.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-fcc80bf83d226889337842b0e4ab887285aa2d2383c75f304b5d1b6640345fa9L32-R34))
*  Use `forkJoin` to combine the observables of corporate cards and card stats, and map them to the `NewCardDetail` interface in `card-stats.component.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-fcc80bf83d226889337842b0e4ab887285aa2d2383c75f304b5d1b6640345fa9L64-R104))
*  Remove the import and declaration of `CorporateCardComponent` from `manage-corporate-cards.module.ts` as it is no longer used in that module ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-5e388261224399599ffa0e4acb99b0c7420e6c17d17c9185be65619fa93d7f31L12),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-5e388261224399599ffa0e4acb99b0c7420e6c17d17c9185be65619fa93d7f31L26-R25))
*  Use a `div` element with the class `manage-corporate-cards__card` to wrap the `app-corporate-card` component and pass the `hideOptionsMenu` input as `true` in `manage-corporate-cards.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-a9990626fe6e417ac0088e9e79ea18f1e0b64e1f644e8cead61870540da85a5eL20-R27))
*  Add the CSS style for the `manage-corporate-cards__card` class in `manage-corporate-cards.page.scss` to set the width and height of the card element ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-b7dfee1b6da49e0de48d7f654d6f1e4d479928ba336a24eb58e1331b0adc2be7R29-R33))
*  Change the text of the button in `add-card.component.html` from "Add Card" to "Add Corporate Card" ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-f9e2659014c7cb2b02a16174e456a776b4765d2278ce91e38b858dc7bc39eacdL7-R7))
*  Rename the `corporate-card` component files from `src/app/fyle/manage-corporate-cards/corporate-card` to `src/app/shared/components/corporate-card` as it is now a shared component ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-1c08c3139af341f98f5dc83524b76783d3ef70c489281b15d00d2da63c1c0f39),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406c))
*  Change the CSS style for the `corporate-card` class in `corporate-card.component.scss` to set the height to 100% instead of 162px ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-89807115c69d39cdf0236632c05103783259617b4795b281ff94b44f05c95d2eL5-R6))
*  Add the input property `hideOptionsMenu` to the class `CorporateCardComponent` in `corporate-card.component.ts` to allow the parent component to control whether to show or hide the card options menu ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-18a2f0b572fcd52e51c9b964dc961c08ab88f1a626d76881dcdf1c42f05f1c3bR14-R15))
*  Check the value of `hideOptionsMenu` input and only show the card options menu if it is false and the card is connected to RTF or statement upload where RTF is enabled in `corporate-card.component.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-18a2f0b572fcd52e51c9b964dc961c08ab88f1a626d76881dcdf1c42f05f1c3bL42-R45))
*  Use the `app-corporate-card` component and pass the `card` input as `cardDetail.card` and the `hideOptionsMenu` input as `true` in `card-detail.component.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff3a492de0c80863621126ba4f3703ec89c04ff6c494c862fe209c6b1b20a099L2-R4))
*  Use the `totalDraftTxns`, `totalDraftValue`, `totalTxnsCount`, and `totalAmountValue` properties of `cardDetail.stats` instead of `cardDetail` to display the number and amount of incomplete and total card expenses in `card-detail.component.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff3a492de0c80863621126ba4f3703ec89c04ff6c494c862fe209c6b1b20a099L38-R10),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff3a492de0c80863621126ba4f3703ec89c04ff6c494c862fe209c6b1b20a099L44-R20),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff3a492de0c80863621126ba4f3703ec89c04ff6c494c862fe209c6b1b20a099L56-R28),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff3a492de0c80863621126ba4f3703ec89c04ff6c494c862fe209c6b1b20a099L62-R39))
*  Fix the formatting of the `humanizeCurrency` pipe to use colons instead of commas as separators in `card-detail.component.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff3a492de0c80863621126ba4f3703ec89c04ff6c494c862fe209c6b1b20a099L44-R20),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff3a492de0c80863621126ba4f3703ec89c04ff6c494c862fe209c6b1b20a099L62-R39))
*  Remove the unused and redundant imports, inputs, and methods from `card-detail.component.ts` and change the type of `cardDetail` input from `CardDetail` to `NewCardDetail` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff87481195f0510b1f6ba2d9438f481221b9668d3ca4cd1d09a0b9d2736129e4L1-R4),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff87481195f0510b1f6ba2d9438f481221b9668d3ca4cd1d09a0b9d2736129e4L13-R13),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff87481195f0510b1f6ba2d9438f481221b9668d3ca4cd1d09a0b9d2736129e4L22-R24),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff87481195f0510b1f6ba2d9438f481221b9668d3ca4cd1d09a0b9d2736129e4L32-R33),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff87481195f0510b1f6ba2d9438f481221b9668d3ca4cd1d09a0b9d2736129e4L43-L44))
*  Remove the redundant HTML elements and CSS styles that were previously used to display the card information in `card-detail.component.html` and `card-detail.component.scss` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-ff3a492de0c80863621126ba4f3703ec89c04ff6c494c862fe209c6b1b20a099L2-R4),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-035721ada299228c0dac25fb45ff677f131554302583ee22caceb110f69438ebL115-R123),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-035721ada299228c0dac25fb45ff677f131554302583ee22caceb110f69438ebL146-L196))
*  Use the `cardDetail` variable instead of `card` to iterate over the `spentCards` array and pass it to the `app-card-detail` component in `spent-cards.component.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-78492bd8c7b8576f817b06758cd4ff3d0b54df406016f5c86896d184221ebeaeL11-R12))
*  Change the CSS style for the `spent-cards--add-card` class in `spent-cards.component.scss` to set the height to 350px instead of 316px ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-5ffddeffb7e5e557ebf833371417195723bbe379cb2654166d6183c3cd6fa22cL9-R9))
*  Remove the unused and redundant imports and methods from `spent-cards.component.ts` and change the type of `spentCards` input from `CardDetail[]` to `NewCardDetail[]` ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-dc99ec76e78f31d3d5f06cf46c9047a33ae6cd5c6611661324231dbbabc0e0a2L1-R5),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-dc99ec76e78f31d3d5f06cf46c9047a33ae6cd5c6611661324231dbbabc0e0a2L15-R16),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-dc99ec76e78f31d3d5f06cf46c9047a33ae6cd5c6611661324231dbbabc0e0a2L26-R29))
*  Import, declare, and export `CorporateCardComponent` in `shared.module.ts` as it is now a shared component ([link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-71ee557103ff1e53a7c4f46a8d510ceb4134b7e22caa6c531c2255b76317100aR144),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-71ee557103ff1e53a7c4f46a8d510ceb4134b7e22caa6c531c2255b76317100aR265),[link](https://github.com/fylein/fyle-mobile-app/pull/2301/files?diff=unified&w=0#diff-71ee557103ff1e53a7c4f46a8d510ceb4134b7e22caa6c531c2255b76317100aR391))

## Clickup
https://app.clickup.com

## UI Preview
Before
image

After
image

### CODE REVIEW

File: src/app/fyle/dashboard/dashboard.service.ts [hunk 1]
```diff
@@ -118,10 +115,6 @@ export class DashboardService {
     };
   }
 
-  getExpenseDetailsInCards(uniqueCards: UniqueCards[], statsResponse: CardAggregateStats[]): UniqueCardStats[] {
```
Comment (OmkarJ13) [inline src/app/fyle/dashboard/dashboard.service.ts:121]: Removed this method from `DashboardService`, in this method we were internally only calling the `CorporateCreditCardExpenseService.getExpenseDetailsInCards` so using this method directly

File: src/app/fyle/manage-corporate-cards/manage-corporate-cards.module.ts [hunk 2]
```diff
@@ -23,6 +22,6 @@ import { CardAddedComponent } from './card-added/card-added.component';
     ManageCorporateCardsPageRoutingModule,
     NgxMaskModule.forRoot(),
   ],
-  declarations: [ManageCorporateCardsPage, CorporateCardComponent, AddCorporateCardComponent, CardAddedComponent],
```
Comment (OmkarJ13) [inline src/app/fyle/manage-corporate-cards/manage-corporate-cards.module.ts:26]: Moved the `CorporateCardComponent` to `SharedModule` to allow use in dashboard as well

File: src/app/shared/components/spent-cards/card-detail/card-detail.component.scss [hunk 3]
```diff
@@ -143,57 +131,6 @@
     box-shadow: 0px -2px 40px rgb(215 215 215 / 40%);
   }
 
-  &--ccc-account-info {
```
Comment (OmkarJ13) [inline src/app/shared/components/spent-cards/card-detail/card-detail.component.scss:146]: Removed unused classes

File: src/app/shared/components/spent-cards/spent-cards.component.html [hunk 4]
```diff
@@ -1,19 +1,19 @@
-<div [ngClass]="{ 'spent-cards': spentCards?.length > 1, 'spent-card': spentCards?.length === 1 }">
```
Comment (OmkarJ13) [inline src/app/shared/components/spent-cards/spent-cards.component.html:1]: The two classes were not needed, as no different stylings are applied. So kept one class only

File: src/app/core/mock-data/card-detail-data.ts
Comment (OmkarJ13) [inline src/app/core/mock-data/card-detail-data.ts:1]: This is not needed anymore, we have added `platform-corporate-card-detail-data.ts`

File: src/app/core/mock-data/card-details.data.ts
Comment (OmkarJ13) [inline src/app/core/mock-data/card-details.data.ts:1]: This was duplicated, we already have the same data in `unique-cards.data.ts`

File: src/app/core/models/card-detail.model.ts
Comment (OmkarJ13) [inline src/app/core/models/card-detail.model.ts:1]: This is duplicated, same model `UniqueCards` already was present

File: src/app/core/services/corporate-credit-card-expense.service.ts [hunk 5]
```diff
@@ -122,13 +123,46 @@ export class CorporateCreditCardExpenseService {
...
+      });
+
+      return formattedCard;
+    });
+  }
+
+  getExpenseDetailsInCards(uniqueCards: UniqueCards[], statsResponse: CardAggregateStats[]): UniqueCardStats[] {
+    const cardsCopy = JSON.parse(JSON.stringify(uniqueCards)) as UniqueCards[];
     const uniqueCardsCopy = [];
-    cardsCopy?.forEach((card: CardDetails) => {
```
Comment (OmkarJ13) [inline src/app/core/services/corporate-credit-card-expense.service.ts:128]: Did some cleanup of types here, the two types `UniqueCards` `CardDetails` and `UniqueCardStats` and `CardDetail` were duplicates of each other. So using only one of them in all places and deleting the duplicate types

File: src/app/fyle/dashboard/card-stats/card-stats.component.html [hunk 6]
```diff
@@ -1,17 +1,17 @@
 <ng-container *ngIf="isConnected$ | async">
   <div *ngIf="isCCCEnabled$ | async" class="card-stats">
-    <ng-container *ngIf="cardTransactionsAndDetails$ | async as cardTransactionsAndDetails">
-      <ng-container *ngIf="cardTransactionsAndDetails?.length > 0">
+    <ng-container *ngIf="cardDetails$ | async as cardDetails">
+      <ng-container *ngIf="cardDetails?.length > 0">
```
Comment (Dimple16) [inline src/app/fyle/dashboard/card-stats/card-stats.component.html:4]: `cardDetails?.length`

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**